pkg/client/set.go

```diff
@@ -25,6 +25,28 @@ import (
 	"sigs.k8s.io/release-utils/version"
 )
 
+// SetInterface provides a unified interface for different Kubernetes clients
+type SetInterface interface {
+
+	// Kubernetes returns the standard Kubernetes clientset
+	Kubernetes() kubernetes.Interface
+
+	// Dynamic returns the dynamic client
+	Dynamic() dynamic.Interface
+
+	// APIExtensionsV1 returns the API extensions client
+	APIExtensionsV1() apiextensionsv1.ApiextensionsV1Interface
+
+	// RESTConfig returns a copy of the underlying REST config
+	RESTConfig() *rest.Config
+
+	// CRD returns a new CRDInterface instance
+	CRD(cfg CRDWrapperConfig) CRDInterface
+
+	// WithImpersonation returns a new client that impersonates the given user
+	WithImpersonation(user string) (SetInterface, error)
+}
+
 // Set provides a unified interface for different Kubernetes clients
 type Set struct {
```

Comment:
should this be made non-exportable ? 
`type set`

Reply:
no, it is fine to export, the constructor returns you a Set* but we store it in controllers as a SetInterface.